x-pack/plugins/ml/public/embeddables/anomaly_explorer/explorer_input_resolver.ts

```diff
@@ -0,0 +1,259 @@
+    distinctUntilChanged(isEqual),
+    switchMap((jobsIds) => anomalyDetectorService.getJobs$(jobsIds)),
+    catchError((e) => {
+      setErrorHandler(e.body ?? e);
+      return of(undefined);
+    })
+  );
+}
+
+export function useExplorerInputResolver(
+  embeddableInput: Observable<AnomalyExplorerEmbeddableInput>,
+  onInputChange: (output: Partial<AnomalyExplorerEmbeddableOutput>) => void,
+  refresh: Observable<any>,
+  services: [CoreStart, MlStartDependencies, AnomalyExplorerServices],
+  chartWidth: number,
+  severity: number
+): { chartsData: ExplorerChartsData; isLoading: boolean; error: Error | null | undefined } {
+  const [
+    { uiSettings },
+    { data: dataServices },
+    { anomalyTimelineService, anomalyDetectorService, anomalyExplorerService },
+  ] = services;
+  const { timefilter } = dataServices.query.timefilter;
+
+  const [chartsData, setChartsData] = useState<any>();
+  const [error, setError] = useState<Error | null>();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const chartWidth$ = useMemo(() => new Subject<number>(), []);
+  const severity$ = useMemo(() => new Subject<number>(), []);
+
+  const timeBuckets = useMemo(() => {
+    return new TimeBuckets({
+      'histogram:maxBars': uiSettings.get(UI_SETTINGS.HISTOGRAM_MAX_BARS),
+      'histogram:barTarget': uiSettings.get(UI_SETTINGS.HISTOGRAM_BAR_TARGET),
+      dateFormat: uiSettings.get('dateFormat'),
+      'dateFormat:scaled': uiSettings.get('dateFormat:scaled'),
+    });
+  }, []);
+
+  useEffect(() => {
+    const subscription = combineLatest([
+      getJobsObservable(embeddableInput, anomalyDetectorService, setError),
+      embeddableInput,
+      chartWidth$.pipe(skipWhile((v) => !v)),
+      severity$,
+      refresh.pipe(startWith(null)),
+    ])
+      .pipe(
+        tap(setIsLoading.bind(null, true)),
+        debounceTime(FETCH_RESULTS_DEBOUNCE_MS),
+        switchMap(([jobs, input, embeddableContainerWidth, severityValue]) => {
+          if (!jobs) {
+            // couldn't load the list of jobs
+            return of(undefined);
+          }
+
+          const { maxSeriesToPlot, timeRange: timeRangeInput, filters, query } = input;
+
+          const viewBySwimlaneFieldName = OVERALL_LABEL;
+
+          anomalyTimelineService.setTimeRange(timeRangeInput);
+
+          const explorerJobs: ExplorerJob[] = jobs.map((job) => {
+            const bucketSpan = parseInterval(job.analysis_config.bucket_span);
+            return {
+              id: job.job_id,
+              selected: true,
+              bucketSpanSeconds: bucketSpan!.asSeconds(),
+            };
+          });
+
+          let influencersFilterQuery: any;
+          try {
+            influencersFilterQuery = processFilters(filters, query);
+          } catch (e) {
+            // handle query syntax errors
+            setError(e);
+            return of(undefined);
+          }
+
+          const bounds = anomalyTimelineService.getTimeBounds();
```

Comment:
I'm not sure if you need to pass an entire `anomalyTimelineService` service.

Reply:
Removed it [here](https://github.com/elastic/kibana/pull/94396/commits/2c6a4d2447b2d87bfa999c62b5b8c36cf9ad890a) and [here](https://github.com/elastic/kibana/pull/94396/commits/6cc91ec3a82adb2ce84fdc1b3b279910480afd02)